components/match2/wikis/ageofempires/match_group_input_custom.lua

```diff
@@ -12,520 +12,342 @@ local Game = require('Module:Game')
-	CustomMatchGroupInput.processPlayerMapData(map, match, 2)
+	-- Store fetched data for following matches
+	Variables.varDefine('tournament_game', data.game)
+	Variables.varDefine('tournament_maps', data.maps)
 
-	match['map' .. mapIndex] = map
+	return match.game or data.game, Logic.emptyOr(mapsInfo, (Json.parse(data.maps)))
 end
 
----@param map table
----@return table
-function CustomMatchGroupInput._mapWinnerProcessing(map)
-	map.scores = {}
-	local hasManualScores = false
-	local indexedScores = {}
-	for scoreIndex = 1, MAX_NUM_OPPONENTS do
-		-- read scores
-		local score = map['score' .. scoreIndex]
-		local obj = {}
-		if Logic.isNotEmpty(score) then
-			hasManualScores = true
-			score = CONVERT_STATUS_INPUT[score] or score
-			if Logic.isNumeric(score) then
-				obj.status = 'S'
-				obj.score = score
-			elseif Table.includes(ALLOWED_STATUSES, score) then
-				obj.status = score
-				obj.score = -1
-			end
-			table.insert(map.scores, score)
-			indexedScores[scoreIndex] = obj
-		else
-			break
-		end
-	end
-
-	local winnerInput = tonumber(map.winner)
-	if Logic.isNotEmpty(map.walkover) then
-		local walkoverInput = tonumber(map.walkover)
-		if walkoverInput == 1 or walkoverInput == 2 or walkoverInput == 0 then
-			map.winner = walkoverInput
-		end
-		map.walkover = Table.includes(ALLOWED_STATUSES, map.walkover) and map.walkover or 'L'
-		map.scores = {-1, -1}
-		map.resulttype = 'default'
+---@param match table
+function CustomMatchGroupInput._getLinks(match)
+	local links = {}
 
-		return map
+	match.civdraft1 = match.civdraft1 or match.civdraft
+	for key, value in Table.iter.pairsByPrefix(match, 'civdraft') do
+		links[key] = 'https://aoe2cm.net/draft/' .. value
 	end
 
-	if hasManualScores then
-		for scoreIndex, _ in Table.iter.spairs(indexedScores, CustomMatchGroupInput._placementSortFunction) do
-			if not tonumber(map.winner) then
-				map.winner = scoreIndex
-			else
-				break
-			end
-		end
-
-		return map
+	match.mapdraft1 = match.mapdraft1 or match.mapdraft
+	for key, value in Table.iter.pairsByPrefix(match, 'mapdraft') do
+		links[key] = 'https://aoe2cm.net/draft/' .. value
 	end
 
-	if map.winner == 'skip' then
-		map.scores = {-1, -1}
-		map.resulttype = 'np'
-	elseif winnerInput == 1 then
-		map.scores = {1, 0}
-	elseif winnerInput == 2 then
-		map.scores = {0, 1}
-	elseif winnerInput == 0 or map.winner == 'draw' then
-		map.scores = {0.5, 0.5}
-		map.resulttype = 'draw'
+	return links
+end
+
+---@param maps table[]
+---@return fun(opponentIndex: integer): integer
+function CustomMatchGroupInput.calculateMatchScore(maps)
+	return function(opponentIndex)
+		return MatchGroupInputUtil.computeMatchScoreFromMapWinners(maps, opponentIndex)
 	end
+end
 
-	return map
+---@param match table
+---@return table
+function CustomMatchGroupInput._getExtraData(match)
+	return {
+		headtohead = Logic.emptyOr(match.headtohead, Variables.varDefault('tournament_headtohead')),
+		civdraft = match.civdraft,
+		mapdraft = match.mapdraft,
```

Comment:
OOS, but i guess this should eventually include multiple drafts input with `civdraft1` etc

Reply:
Is this acutally needed in extradata? Considering the links are in links

Reply:
If links are available from the API, no, we could drop it entirely then

Reply:
Yeah they are, within `match2.links`